app/models/name/synonymy.rb

```diff
@@ -215,4 +215,28 @@ def time_of_last_naming
       last_use || created_at
     end
   end
+
+  # "Best" preferred synonym of a deprecated name.
+  def best_preferred_synonym
+    most_recently_updated(preferred_synonyms_with_most_observations)
```

Comment:
The algorithm here is slightly different from that in `more_popular` (used by EOL?).
`more_popular` picks the Name which has more Observations or was last used or was last created.
`best_preferred_synonym` picks the Name which has more Observations or was last updated. That's simpler to code, and doesn't hit the db.